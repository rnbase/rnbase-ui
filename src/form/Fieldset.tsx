import React from 'react';
import { StyleSheet, View, ViewProps } from 'react-native';

import { Themed, withTheme } from '../theming';

interface FieldsetProps extends ViewProps {}

const Fieldset: React.FC<Themed<typeof createStyleSheet, FieldsetProps>> = ({
  theme: { styles },
  children,
  style,
  ...props
}) => (
  <View {...props} style={[styles.root, style]}>
    {children}
  </View>
);

const createStyleSheet = () =>
  StyleSheet.create({
    root: {
      marginVertical: 15,
    },
  });

export default withTheme(Fieldset, createStyleSheet, 'Form.Fieldset');
