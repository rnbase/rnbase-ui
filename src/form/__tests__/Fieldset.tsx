import React from 'react';

// Note: test renderer must be required after react-native.
import TestRenderer from 'react-test-renderer';

import Fieldset from '../Fieldset';

type Props = React.ComponentProps<typeof Fieldset>;

const Row = () => React.createElement('Row');

/**
 * Under test
 */
const createElement = (props: Partial<Props>) => (
  <Fieldset {...props}>
    <Row />
  </Fieldset>
);

const createRenderer = (props: Partial<Props>) => TestRenderer.create(createElement(props));

it('should render normally', () => {
  expect(createRenderer({})).toMatchSnapshot();
});

it('should render with custom styles', () => {
  expect(createRenderer({ style: { height: 123 } })).toMatchSnapshot();
});
