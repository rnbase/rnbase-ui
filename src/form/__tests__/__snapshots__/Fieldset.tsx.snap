// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render normally 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 15,
      },
      undefined,
    ]
  }
>
  <Row />
</View>
`;

exports[`should render with custom styles 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 15,
      },
      Object {
        "height": 123,
      },
    ]
  }
>
  <Row />
</View>
`;
